src/PrestaShopBundle/Controller/Admin/Sell/Catalog/FeatureController.php

```diff
@@ -0,0 +1,178 @@
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Controller\Admin\Sell\Catalog;
+
+use Exception;
+use PrestaShop\PrestaShop\Core\Domain\Feature\Exception\FeatureConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Feature\Query\GetFeatureForEditing;
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Controller responsible for "Sell > Catalog > Attributes & Features > Features" page
+ */
+class FeatureController extends FrameworkBundleAdminController
+{
+    /**
+     * Create feature action.
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function createAction(Request $request)
+    {
+        if (!$this->isFeatureEnabled()) {
+            return $this->render('@PrestaShop/Admin/Sell/Catalog/Features/create.html.twig', [
+                'showDisabledFeatureWarning' => true,
+            ]);
+        }
+
+        $featureFormBuilder = $this->get('prestashop.core.form.identifiable_object.builder.feature_form_builder');
+        $featureFormHandler = $this->get('prestashop.core.form.identifiable_object.handler.feature_form_handler');
+
+        $featureForm = $featureFormBuilder->getForm();
+        $featureForm->handleRequest($request);
+
+        try {
+            $handlerResult = $featureFormHandler->handle($featureForm);
+
+            if (null !== $handlerResult->getIdentifiableObjectId()) {
+                $this->addFlash('success', $this->trans('Successful creation.', 'Admin.Notifications.Success'));
+
+                //@todo change route to index when it's migrated
+                return $this->redirectToRoute('admin_features_create');
+            }
+        } catch (Exception $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+        }
+
+        return $this->render('@PrestaShop/Admin/Sell/Catalog/Features/create.html.twig', [
+            'featureForm' => $featureForm->createView(),
+        ]);
+    }
+
+    /**
+     * Edit feature action.
+     *
+     * @param int $featureId
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function editAction($featureId, Request $request)
+    {
+        try {
+            $editableFeature = $this->getQueryBus()->handle(new GetFeatureForEditing((int) $featureId));
+        } catch (Exception $e) {
```

Comment:
What about `NotFoundException` if `$featureId` does not exist ?

Maybe add something like

```
            if ($e instanceof FeatureNotFoundException) {
                return $this->redirectToRoute(...);
            }
```

Reply:
it needs to redirect in case of any exception, but now I've fixed it to show a specific error message if the feature was not found